js/PuzzleOne.js

```diff
@@ -0,0 +1,354 @@
+  DIA1.replaceWith(DIA5);
+  DIA2.replaceWith(DIA5);
+  DIA3.replaceWith(DIA5);
+  DIA4.replaceWith(DIA5);
+  DIA6.replaceWith(DIA5);
+
+  DIA5.classList.remove("hidden");
+
+});
+
+// PUZZLE 1 SUBMIT ANSWER BUTTON
+// On click compares answer from user input element (a1) to (P1A)
+SA1Btn.addEventListener("click", () => {
+
+  // If the answer is correct, reveals next puzzle Nav Button
+  if (a1.value == P1A) {
+
+    console.log("haha its correct!");
+    NAV2.classList.add("reveal");
+    alert("Check Navigation tab for new puzzle!");
+
+    // If the answer is wrong, alerts the user it's wrong
+  } else {
+
+    console.log("Sorry, try again!");
+    alert("I don't think that's the answer");
+
+  }
+});
+
+// PUZZLE 2 SUBMIT ANSWER BUTTON
+// On click compares answer from user input element (a2) to (P2A)
+SA2Btn.addEventListener("click", () => {
+  // If the answer is correct, reveals next puzzle Nav Button
+  if (a2.value == P2A) {
+
+    console.log("haha its correct!");
+    NAV3.classList.add("reveal");
+    alert("Check Navigation tab for new puzzle!");
+
+    // If the answer is wrong, alerts the user it's wrong
+  } else {
+
+    console.log("Sorry, try again!");
+    alert("I don't think that's the answer");
+
+  }
+});
+
+// PUZZLE 3 SUBMIT ANSWER BUTTON
+// On click compares answer from user input element (a3) to (P3A)
+SA3Btn.addEventListener("click", () => {
+  // If the answer is correct, reveals next puzzle Nav Button
+  if (a3.value == P3A) {
+
+    console.log("haha its correct!");
+    NAV4.classList.add("reveal");
+    alert("Check Navigation tab for new puzzle!");
+
+    // If the answer is wrong, alerts the user it's wrong
+  } else {
+    console.log("Sorry, try again!");
+    alert("I don't think that's the answer");
+
+  }
+});
+
+// PUZZLE 4 REVEAL THE PASSWORD BUTTON
+// Clicking the text "button" in the <p> element, will bring up an alert and show the password
+reveal_button.addEventListener("click", () => {
+
+  console.log("He clicked it pog!");
+  alert("The password is chenga");
+
+});
+
+// PUZZLE 4 SUBMIT ANSWER BUTTON
+// On click compares answer from user input element (a4) to (P4A)
+SA4Btn.addEventListener("click", () => {
+
+  // If the answer is correct, reveals next puzzle Nav Button
+  if (a4.value == P4A) {
+
+    console.log("haha its correct!");
+    NAV5.classList.add("reveal");
+    Danger1.classList.remove("hidden");
+    alert("Check Navigation tab for new puzzle!");
+
+    // If the answer is wrong, alerts the user it's wrong
+  } else {
+    console.log("Sorry, try again!");
+    alert("I don't think that's the answer");
+
+  }
+});
+
+// PUZZLE 5
+// if you click yes it will open an Alert asking to confirm
+// clicking no will open a youtube link to an angry opossum
+SA5Btn_Accept.addEventListener("click", () => {
+
+  console.log("ay it works!");
+
+  // confirming again, will open a youtube link and change DIAX element to DIA6
+  if (confirm("ARE YOU SURE? KIDNEYS DONT GROW BACK!") == true) {
```

Comment:
Koja je razlika izmedju `==` i `===` u JavaScriptu?

Reply:
> Koja je razlika izmedju `==` i `===` u JavaScriptu?

`==` upoređuje samo value

`===` upoređuje i value i type. Što znači da ako upoređujemo dve varijable rezultat će biti `true` samo ako upoređene varijable imaju istu vrednost i isti tip vrednosti. 

Npr:

`a1 = 69`

`a1 === 420` -> `false`
`a1 === 69` -> `true` jer se poklapaju vrednosti i tip vrednosti
`a1 === "69"` -> `false` jer se ne poklapaju tip vrednosti. `"69"` je string, dok `a1` ima definisan tip vrednosti kao  `number`


____

`==` poređenje funkcioniše isto i za Boolean tip ali se za number/string koriste `(0, 1)` vrednosti
`b2 = true`

`b2 == false` - `false` rezultat
`b2 == true` - `true` rezultat
`b2 == "1"` - `true` rezultat
`b2 == 1` - `true` rezultat